mindsdb/integrations/handlers/reddit_handler/reddit_handler.py

```diff
@@ -0,0 +1,296 @@
+    HandlerStatusResponse as StatusResponse,
+    HandlerResponse as Response,
+    RESPONSE_TYPE
+)
+from mindsdb.utilities.config import Config
+from mindsdb.utilities import log
+from mindsdb_sql.parser import ast
+from mindsdb.integrations.utilities.sql_utils import extract_comparison_conditions
+from mindsdb.integrations.utilities.date_utils import parse_utc_date
+
+from mindsdb_sql.parser.ast import Select
+
+
+class CommentTable(APITable):
+    def select(self, query: ast.Select) -> pd.DataFrame:
+        '''Select data from the comment table and return it as a pandas DataFrame.
+
+        Args:
+            query (ast.Select): The SQL query to be executed.
+
+        Returns:
+            pandas.DataFrame: A pandas DataFrame containing the selected data.
+        '''
+
+        reddit = self.handler.connect()
+
+        submission_id = None
+        conditions = extract_comparison_conditions(query.where)
+        for condition in conditions:
+            if condition[0] == '=' and condition[1] == 'submission_id':
+                submission_id = condition[2]
+                break
+
+        if submission_id is None:
+            raise ValueError('Submission ID is missing in the SQL query')
+
+        submission = reddit.submission(id=submission_id)
+        submission.comments.replace_more(limit=None)
+
+        result = []
+        for comment in submission.comments.list():
+            data = {
+                'id': comment.id,
+                'body': comment.body,
+                'author': comment.author.name if comment.author else None,
+                'created_utc': comment.created_utc,
+                'score': comment.score,
+                'permalink': comment.permalink,
+                'ups': comment.ups,
+                'downs': comment.downs,
+                'subreddit': comment.subreddit.display_name,
+            }
+            result.append(data)
+
+        result = pd.DataFrame(result)
+        self.filter_columns(result, query)
+        return result
+
+
+    def get_columns(self):
+        '''Get the list of column names for the comment table.
+
+        Returns:
+            list: A list of column names for the comment table.
+        '''
+        return [
+            'id',
+            'body',
+            'author',
+            'created_utc',
+            'permalink',
+            'score',
+            'ups',
+            'downs',
+            'subreddit',
+        ]
+
+    def filter_columns(self, result: pd.DataFrame, query: ast.Select = None):
+        columns = []
+        if query is not None:
+            for target in query.targets:
+                if isinstance(target, ast.Star):
+                    columns = self.get_columns()
+                    break
+                elif isinstance(target, ast.Identifier):
+                    columns.append(target.value)
+        if len(columns) > 0:
+            result = result[columns]
+
+class SubmissionTable(APITable):
```

Comment:
Can we please move this class to a new file as reddit_tables.py and keep only the handler here?